### PR TITLE
Add page subtitle

### PR DESCRIPTION
**Pull request by JohnnyMaynne**
Adding a subtitle to a page.

```
class ListPages extends ListRecords
{ 
    protected static ?string $subtitle = 'Anim aute id magna aliqua ad ad non deserunt sunt.';
}
```

image

### CODE REVIEW

File: packages/admin/resources/views/components/header/heading.blade.php [hunk 1]
```diff
@@ -1,3 +1,3 @@
-<h1 {{ $attributes->class(['text-2xl font-bold tracking-tight md:text-3xl filament-header-heading']) }}>
+<h1 {{ $attributes->class(['text-2xl font-bold tracking-tight filament-header-heading']) }}>
```
Comment (zepfietje) [inline packages/admin/resources/views/components/header/heading.blade.php]: Unrelated to this PR, but I'll update it for consistency.

```suggestion
<h1 {{ $attributes->class(['filament-header-heading text-2xl font-bold tracking-tight']) }}>
```

File: packages/admin/resources/views/components/header/subheading.blade.php [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<h2 {{ $attributes->class(['text-gray-500 tracking-tight filament-header-subheading max-w-2xl']) }}>
```
Comment (zepfietje) [inline packages/admin/resources/views/components/header/subheading.blade.php]:
```suggestion
<h2 {{ $attributes->class(['filament-header-subheading max-w-2xl tracking-tight text-gray-500']) }}>
```

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/admin/resources/views/components/header/index.blade.php [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
 <header {{ $attributes->class(['filament-header space-y-2 items-start justify-between sm:flex sm:space-y-0 sm:space-x-4  sm:rtl:space-x-reverse sm:py-4']) }}>
-    <x-filament::header.heading>
-        {{ $heading }}
-    </x-filament::header.heading>
+    <div>
+        <x-filament::header.heading>
+            {{ $heading }}
+        </x-filament::header.heading>
+
+        @if($subheading)
```
Comment (zepfietje) [inline packages/admin/resources/views/components/header/index.blade.php]:
```suggestion
        @if ($subheading)
```

File: packages/admin/resources/views/components/page.blade.php [hunk 4]
```diff
@@ -12,9 +12,15 @@
                 <x-slot name="heading">
                     {{ $heading }}
                 </x-slot>
+
+                @if($subheading = $this->getSubheading())
```
Comment (zepfietje) [inline packages/admin/resources/views/components/page.blade.php]:
```suggestion
                @if ($subheading = $this->getSubheading())
```

File: packages/admin/src/Pages/Page.php [hunk 5]
```diff
@@ -198,6 +200,11 @@ protected function getTitle(): string
             ->title();
     }
 
+    protected function getSubheading(): ?string
+    {
+        return static::$subtitle;
```
Comment (zepfietje) [inline packages/admin/src/Pages/Page.php]: I'm not sure about the naming of the property.

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/admin/src/Pages/Page.php (hunk 5 above)
Comment (Z3d0X) [reply packages/admin/src/Pages/Page.php]: Maybe rename it to "description" instead of subtitle/subheading

Comment (zepfietje) [reply packages/admin/src/Pages/Page.php]: Might be confusing with meta description. I think subheading could be suitable actually. Just not sure about the internal subtitle prop.

Comment (danharrin) [reply packages/admin/src/Pages/Page.php]: Yeah this needs to be `$subheading`

**Pull request merged**